src/elements/date-input/date-input.component.ts

```diff
@@ -70,23 +67,37 @@ class SbbDateInputElement<T = Date> extends SbbFormAssociatedInputMixin(LitEleme
     ) {
       // Align with the native date input, as it copies the value of
       // the given date and does not retain the original instance.
-      this._valueAsDate = value;
+      this._valueAsDate = this._dateAdapter.createDate(
+        this._dateAdapter.getYear(value),
+        this._dateAdapter.getMonth(value),
+        this._dateAdapter.getDate(value),
+      );
       const stringValue = this._formatDate();
       this._valueCache = [stringValue, this._valueAsDate];
       this.value = stringValue;
     }
   }
   public get valueAsDate(): T | null {
-    return this._valueAsDate ?? null;
+    // We should never return the internal date object directly, as it
+    // could be manipulated outside of this class.
+    return this._valueAsDate ? this._dateAdapter.clone(this._valueAsDate) : null;
   }
   private _valueAsDate?: T | null;
 
-  @forceType()
-  @property({ converter: dateConverter, reflect: true, type: DateOnlyType })
+  /**
+   * The minimum valid date. Accepts a date object or null.
+   * Accepts an ISO8601 formatted string (e.g. 2024-12-24) as attribute.
+   */
+  @plainDate()
+  @property({ converter: plainDateConverter, reflect: true })
```

Comment:
Why do we have to reflect the date?

Reply:
Parity with the native input. Probably not important, but I prefer it this way.
Or what is your opinion?

Reply:
I think I maybe understand it now, because the now property of the calendarElement has a fallback, we can assign null and it fallbacks to today(); but in theory now doesn't allow null.

Reply:
ah ok, then let's keep it. I just thought probably we never use the attribute.